server/app/build.gradle

```diff
@@ -61,53 +61,117 @@ task generateActionsList {
         outputDirectory.mkdirs()
         outputFile.createNewFile()
 
-        println("Generate action list:")
+        println("=== Generating Action List ===")
         outputFile.withWriter {
             writer ->
                 actions.each {
                     action -> writer.writeLine action
                     println(action)
                 }
         }
-    }
-}
 
-task prepareAssets(type: Copy) {
-    from "${exampleManifestFile}", "${calabashJs}", "${versionFilePath}"
-    into "${assetsDir}"
+        println("=== Adding Additional Assets ===")
+        println("Adding ${exampleManifestFile}")
+        exec {
+            commandLine "cp", "${exampleManifestFile}", "${assetsDir}"
+        }
+        println("Adding ${calabashJs}")
+        exec {
+            commandLine "cp", "${calabashJs}", "${assetsDir}"
+        }
+        println("Adding ${versionFilePath}")
+        exec {
+            commandLine "cp", "${versionFilePath}", "${assetsDir}"
+        }
+    }
 }
 
 task prepareApk {
     doLast {
+        def apk = "${outputPath}${apkName}"
+        def server = "${outputPath}${serverApkName}"
+
         println("Stripping signature and manifest from server apk...")
 
         exec {
-            commandLine "zip", "-q", "-d", "${outputPath}${apkName}", "META-INF/*"
+            commandLine "zip", "-q", "-d", "${apk}", "META-INF/*"
         }
         exec {
-            commandLine "zip", "-q", "-d", "${outputPath}${apkName}", "AndroidManifest.xml"
+            commandLine "zip", "-q", "-d", "${apk}", "AndroidManifest.xml"
         }
+
+        exec {
+            commandLine "rm", "-f", "${server}"
+        }
+
         exec {
-            commandLine "mv", "${outputPath}${apkName}", "${outputPath}${serverApkName}"
+            commandLine "cp", "${apk}", "${server}"
         }
 
-        println("Build done.")
-        println("Output ready in: ${outputPath}${serverApkName}")
+
+        println("Done building.")
+        println("Verifying the apk...")
+
+
+        def tmpDir = new File("../../tmp/verify-apk")
+        tmpDir.delete()
+        tmpDir.mkdirs()
+
+        ant.unzip(src: "${server}", dest: tmpDir.absolutePath)
+        def files = []
+
+        // 'it' is a string until it is added to files, then it is a File
+        tmpDir.traverse(type: groovy.io.FileType.FILES) { files.add(it) }
```

Comment:
when there is confusion on the type of "it" its recommended to use parameter override:

```groovy
tmpDir.traverse(type: groovy.io.FileType.FILES) {fileName -> files.add(fileName) }
files = files.collect {file -> file.path}
``` 

With that said, then traverse should be returning file objects... (http://docs.groovy-lang.org/latest/html/documentation/working-with-io.html#_traversing_file_trees)